ssh-server/main.go

```diff
@@ -0,0 +1,149 @@
+		if err != nil {
+			log.Fatalf("Failed to accept incoming connection: %v", err)
+		}
+
+		go func() {
+			handleConnection(connection, config)
+		}()
+	}
+}
+
+func startServer(port uint) (*ssh.ServerConfig, net.Listener, error) {
+	config, err := getConfiguration()
+	if err != nil {
+		return nil, nil, fmt.Errorf("getting configuration: %v", err)
+	}
+
+	serverAddress := fmt.Sprintf(":%d", port)
+	listener, err := net.Listen("tcp", serverAddress)
+	if err != nil {
+		return nil, nil, fmt.Errorf("listen: %v", err)
+	}
+	return config, listener, nil
+}
+
+func getConfiguration() (*ssh.ServerConfig, error) {
+	config := &ssh.ServerConfig{NoClientAuth: true}
+
+	privateBytes, err := ioutil.ReadFile("id_rsa")
+	if err != nil {
+		return nil, fmt.Errorf("reading private server key: %v", err)
+	}
+	private, err := ssh.ParsePrivateKey(privateBytes)
+	if err != nil {
+		return nil, fmt.Errorf("parsing private server key: %v", err)
+	}
+	config.AddHostKey(private)
+	return config, nil
+}
+
+func handleConnection(conn net.Conn, serverConfig *ssh.ServerConfig) error {
+	_, chans, reqs, err := ssh.NewServerConn(conn, serverConfig)
+	if err != nil {
+		return fmt.Errorf("handshake: %v", err)
+	}
+
+	// Discard out-of-band SSH requests (not supported by this server).
+	go ssh.DiscardRequests(reqs)
+
+	for newChannel := range chans {
+		if err := serviceChannel(newChannel); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func serviceChannel(newChannel ssh.NewChannel) error {
+	if newChannel.ChannelType() != "session" {
+		newChannel.Reject(ssh.UnknownChannelType, "unknown channel type")
+		return nil
+	}
+	channel, requests, err := newChannel.Accept()
+	if err != nil {
+		return fmt.Errorf("accepting channel: %v", err)
+	}
+	defer channel.Close()
+
+	// Start the command with a pseudo-terminal.
+	bashPTY, err := pty.Start(exec.Command("bash"))
+	if err != nil {
+		return fmt.Errorf("starting pty: %v", err)
+	}
+	defer bashPTY.Close()
+
+	go func(in <-chan *ssh.Request) {
+		for req := range in {
+			// Tell the client we accept pty and shell commands
+			req.Reply(req.Type == "pty-req" || req.Type == "shell", nil)
+
+			if req.Type == "pty-req" {
+				skip := binary.BigEndian.Uint32(req.Payload)
+				if err := resize(bashPTY, req.Payload[4+skip:]); err != nil {
+					channel.Stderr().Write([]byte(fmt.Sprintf("Failed to resize pty: %v. ", err)))
+					channel.Close()
+					return
+				}
+			}
+			if req.Type == "window-change" {
+				if err := resize(bashPTY, req.Payload); err != nil {
+					channel.Stderr().Write([]byte(fmt.Sprintf("Failed to resize pty: %v. ", err)))
+					channel.Close()
+					return
+				}
+			}
+		}
+	}(requests)
+
+	go io.Copy(bashPTY, channel)
+	io.Copy(channel, bashPTY)
+	return nil
+}
+
+func resize(bashPTY *os.File, payload []byte) error {
+	var winSize struct {
+		Width  uint32
+		Height uint32
+	}
+
+	buf := bytes.NewReader(payload)
+	err := binary.Read(buf, binary.BigEndian, &winSize)
```

Comment:
Just inline the reader since you don't need it elsewhere

if err := binary.Read(bytes.NewReader(payload), binary.BigEndian, &size); err == nil {
  pty.SetSize(f, size)
}

Reply:
done